### PR TITLE
✨[amp-story] Disable first page animations when `#embedMode=3`

### PR DESCRIPTION
**Pull request by calebcordry**
Allows viewers setting this param to disable animation. Referred to as `EmbedMode.PREVIEW` in the codebase.

### CODE REVIEW

File: extensions/amp-story/1.0/animation.js [hunk 1]
```diff
@@ -545,11 +546,15 @@ export class AnimationManager {
     /** @private @const */
     this.builderPromise_ = this.createAnimationBuilderPromise_();
 
+    const storeService = getStoreService(this.ampdoc_.win);
+    const animationDisabled =
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/animation.js:550]: We can rename this var to `firstPageAnimationDisabled` to make it clear that it only impacts the first page

**Review by mszylkowski — APPROVED** [review]
Aapproving with 1 nit

File: extensions/amp-story/1.0/amp-story-store-service.js [hunk 2]
```diff
@@ -120,6 +120,7 @@ export let State;
 /** @const @enum {string} */
 export const StateProperty = {
   // Embed options.
+  CAN_ANIMATE_FIRST_PAGE: 'canAnimateFirstPage',
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-store-service.js:123]: Nit:
```suggestion
  CAN_RUN_ANIMATIONS_ON_INITIAL_PAGE: ...,
```

Would be a bit more accurate

File: extensions/amp-story/1.0/amp-story.js [hunk 3]
```diff
@@ -402,6 +402,7 @@ export class AmpStory extends AMP.BaseElement {
     }
     if (
       isExperimentOn(this.win, 'story-disable-animations-first-page') ||
+      !this.storeService_.get(StateProperty.CAN_ANIMATE_FIRST_PAGE) ||
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.js:405]: The store service should always have the correct and up to date value, here this state would say `true` even when the experiment on the line above is enabled

Comment (calebcordry) [reply extensions/amp-story/1.0/amp-story.js:405]: Are you saying this should be something like
```js
if (isExperimentOn(this.win, 'story-disable-animations-first-page') {
  this.storeService.set(StateProperty.CAN_ANIMATE_FIRST_PAGE, false)
}
```
then always check the store value?

This means we are ok with the experiment val overriding the `embedMode` default?

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js:405]: No we don't need to change the UX, but the value from the store service should always be the one source of truth.
We should remove all other `isExperimentOn()` and `prefersReducedMotion` checks and only rely on the store service, which is why it should always be up to date with the right value.

File: extensions/amp-story/1.0/amp-story-store-service.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-story/1.0/amp-story-store-service.js]: Removing this after our chat.

File: extensions/amp-story/1.0/amp-story.js (hunk 3 above)
Comment (calebcordry) [reply extensions/amp-story/1.0/amp-story.js]: Changing this as well.

File: extensions/amp-story/1.0/animation.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-story/1.0/animation.js]: Done.

**Comment by calebcordry** [comment]
Refactored to not use store and instead use a helper function that tells us if we are in preview mode. PTAL

**Comment by mszylkowski** [comment]
What exactly do we want to mean by "preview mode"? I don't understand how the preview mode relates to the animation not playing (naming-wise).

**Review by gmajoulet — APPROVED** [review]

**Pull request merged**